fgrpc/grpcrunner.go

```diff
@@ -43,17 +43,30 @@ const (
 	prefixHTTPS      = "https://"
 )
 
-// Dial dials grpc either using insecure or using default tls setup.
-// TODO: option to specify certs.
-func Dial(serverAddr string, tls bool) (conn *grpc.ClientConn, err error) {
-	opts := grpc.WithInsecure()
-	if tls || (strings.HasPrefix(serverAddr, "https://")) {
-		opts = grpc.WithTransportCredentials(credentials.NewTLS(nil))
+// Dial dials grpc using insecure or tls transport security when serverAddr
+// has prefixHTTPS or cert is provided. If override is set to a non empty string,
+// it will override the virtual host name of authority in requests.
+func Dial(serverAddr string, cacert, override string) (conn *grpc.ClientConn, err error) {
+	var opts []grpc.DialOption
+	switch {
+	case cacert != "":
+		creds, err := credentials.NewClientTLSFromFile(cacert, override)
+		if err != nil {
+			log.Errf("Invalid TLS credentials: %v\n", err)
+			return nil, err
+		}
+		log.Infof("Using CA certificate %v to construct TLS credentials", cacert)
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	case strings.HasPrefix(serverAddr, prefixHTTPS):
+		creds := credentials.NewTLS(nil)
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	default:
+		opts = append(opts, grpc.WithInsecure())
 	}
 	serverAddr = grpcDestination(serverAddr)
-	conn, err = grpc.Dial(serverAddr, opts)
+	conn, err = grpc.Dial(serverAddr, opts...)
 	if err != nil {
-		log.Errf("failed to conect to %s with tls %v: %v", serverAddr, tls, err)
+		log.Errf("failed to connect to %s: %v", serverAddr, err)
```

Comment:
NeedFix: when an error occurs it is very important to log all the input, please add cacert and override to the statement